src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,197 @@
+const BASE_GAUGE_GENERATION_AMOUNT = 10
+
+/**
+ * Base amount for the soul gauge consumption
+ */
+const BASE_GAUGE_CONSUMPTION_AMOUNT = 50
+const PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT = 50
+const ENSHROUD_GAUGE_CONSUMPTION = 50
+
+/**
+ * Amount generated for the "Soul" skills, which are cooldowns
+ */
+const SOUL_GAUGE_GENERATION_AMOUNT = 50
+
+export class Gauge extends CoreGauge {
+	@dependency private suggestions!: Suggestions
+
+	private soulGeneratorModifiers = new Map<number, number>([
+		[this.data.actions.SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.WAXING_SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.INFERNAL_SLICE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SPINNING_SCYTHE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.NIGHTMARE_SCYTHE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SOUL_SLICE.id, SOUL_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.SOUL_SCYTHE.id, SOUL_GAUGE_GENERATION_AMOUNT],
+	])
+
+	private soulConsumptionModifiers = new Map<number, number>([
+		[this.data.actions.BLOOD_STALK.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.GRIM_SWATHE.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.GLUTTONY.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.UNVEILED_GIBBET.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+		[this.data.actions.UNVEILED_GALLOWS.id, BASE_GAUGE_CONSUMPTION_AMOUNT],
+	])
+
+	private shroudGeneratorModifiers = new Map<number, number>([
+		[this.data.actions.GIBBET.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.GALLOWS.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.GUILLOTINE.id, BASE_GAUGE_GENERATION_AMOUNT],
+		[this.data.actions.PLENTIFUL_HARVEST.id, PLENTIFUL_HARVEST_SHROUD_GENERATION_AMOUNT],
+	])
+
+	private shroudConsumerModifiers = new Map<number, number>([
+		[this.data.actions.ENSHROUD.id, ENSHROUD_GAUGE_CONSUMPTION],
+	])
+
+	private soulGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SOUL_MAX_VALUE,
+		graph: {
+			label: 'Soul Gauge',
+			color: SOUL_GAUGE_COLOR.fade(0.25),
+		},
+		correctHistory: true,
+	}))
+
+	private shroudGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SHROUD_MAX_VALUE,
+		graph: {
+			label: 'Shroud Gauge',
+			color: SHROUD_GAUGE_COLOR.fade(0.25),
+		},
+		correctHistory: true,
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		// soul gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS)), this.onSoulGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_CONSUMERS)), this.onSoulConsumption)
+
+		// shroud gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_GENERATORS)), this.onShroudGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_CONSUMERS)), this.onShroudConsumption)
```

Comment:
I would set this up as:

- Hook for things that generate 10 soul gauge (default behavior)
- Hook for things that generate 50 soul gauge (edge case)
- Hook for consumers
// shroud gauge
- Hook for things that generate 10 shroud Gauge
- Hook for Plentiful Harvest (edge case)
- Hook for Enshroud

It's 2 extra hooks, but it'll make your code way cleaner than having all of the extra noise involved in accepting and handling the maps you're using right now.